improver/wxcode/utilities.py

```diff
@@ -200,35 +198,22 @@ def interrogate_decision_tree(wxtree):
 
     # Diagnostic names and threshold values.
     requirements = {}
-    # How the data has been thresholded relative to these thresholds.
-    relative = {}
 
     for query in queries.values():
         diagnostics = expand_nested_lists(query, 'diagnostic_fields')
-        for index, diagnostic in enumerate(diagnostics):
-            if diagnostic not in requirements:
-                requirements[diagnostic] = []
-                relative[diagnostic] = []
-            requirements[diagnostic].extend(
-                expand_nested_lists(query, 'diagnostic_thresholds')[index])
-            relative[diagnostic].append(
-                expand_nested_lists(query, 'diagnostic_conditions')[index])
+        thresholds = expand_nested_lists(query, 'diagnostic_thresholds')
+        for diagnostic, threshold in zip(diagnostics, thresholds):
+            unique_thresholds = requirements.setdefault(diagnostic, [])
+            if threshold not in unique_thresholds:
+                unique_thresholds.append(threshold)
 
     # Create a list of formatted strings that will be printed as part of the
     # CLI help.
     output = []
-    for requirement in requirements:
-        entries = np.array([entry for entry in requirements[requirement]])
-        relations = np.array([entry for entry in relative[requirement]])
-        _, thresholds = np.unique(np.array([item.points.item()
-                                            for item in entries]),
-                                  return_index=True)
-        output.append('{}; thresholds: {}'.format(
-            requirement, ', '.join([
-                '{} ({})'.format(str(threshold.points.item()),
-                                 str(threshold.units))
-                for threshold, relation in
-                zip(entries[thresholds], relations[thresholds])])))
+    for requirement, unique_thresholds in sorted(requirements.items()):
+        units, = set(u for (_, u) in unique_thresholds)  # enforces same units
+        thresh_str = ', '.join(str(v) for (v, _) in unique_thresholds)
```

Comment:
Should `unique_thresholds` be sorted in the displayed extended help?

Reply:
That would be good. It previously created them in the defined order, which happens to be sorted. This change means that is no longer the case, so sorting seems sensible.